joblib/my_exceptions.py

```diff
@@ -10,23 +10,22 @@
 
 class JoblibException(Exception):
     """A simple exception with an error message that you can get to."""
-
     def __init__(self, *args):
-        self.args = args
-
-    def __reduce__(self):
-        # For pickling
-        return self.__class__, self.args, {}
+        # We need to implement __init__ so that it is picked in the
+        # multiple heritance hierarchy in the class created in
+        # _mk_exception. Note: in Python 2, if you implement
+        # __init__ in your exception class you need to set .args
+        # correctly, otherwise you can dump an exception instance with
+        # pickle but not load it (at load time an empty .args will be
+        # passed to the constructor)
+        Exception.__init__(self, *args)
```

Comment:
I really want to call `Exception.__init__` here in order to set .args as I am trying to explain in the comment.

Using super cause the second base class `__init__` to be called (in this case the exception that happens in the worker process). We have no control on its constructor signature so we probably don't want to do that.

Here is a snippet to try to make things clearer:
```python
class Base1(object):
    def __init__(self):
        print('Base1')
        super(Base1, self).__init__()


class Base2(object):
    def __init__(self):
        print('Base2')


class Derived(Base1, Base2):
    pass

d = Derived()
```

Output:
```
Base1
Base2
```